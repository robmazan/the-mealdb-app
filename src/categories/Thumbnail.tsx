import React from "react";
import { Link } from "react-router-dom";
import styles from "./Thumbnail.module.scss";

type ThumbnailProps = {
  to: string;
  label: string;
  tooltip?: string;
  imgSrc: string;
  imgAlt: string;
  imgWidth?: number;
  imgHeight?: number;
};

const Thumbnail: React.FC<ThumbnailProps> = ({
  to,
  label,
  tooltip,
  imgSrc,
  imgAlt,
  imgWidth,
  imgHeight
}) => {
  const dimensions = {
    ...(imgWidth ? { width: `${imgWidth}px` } : {}),
    ...(imgHeight ? { height: `${imgHeight}px` } : {})
  };
  return (
    <Link className={styles.link} title={tooltip} to={to}>
      <img className={styles.thumb} src={imgSrc} alt={imgAlt} {...dimensions} />
      {label}
    </Link>
  );
};

export default Thumbnail;
