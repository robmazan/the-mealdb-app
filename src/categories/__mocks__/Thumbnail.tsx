import React from "react";

const Thumbnail: React.FC = props => {
  return <code>{JSON.stringify(props)}</code>;
};

export default Thumbnail;
