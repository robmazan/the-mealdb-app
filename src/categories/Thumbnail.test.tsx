import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { StaticRouter } from "react-router";
import Thumbnail from "./Thumbnail";

it("renders without crashing", () => {
  const container = document.createElement("div");
  act(() => {
    render(
      <StaticRouter>
        <Thumbnail
          label="Label"
          imgAlt="Alt"
          imgSrc="https://example.com/img.jpg"
          to="/path/to"
          tooltip="tooltip"
          imgWidth={100}
          imgHeight={100}
        />
      </StaticRouter>,
      container
    );
  });
  expect(container.innerHTML).toMatchInlineSnapshot(
    `"<a class=\\"link\\" title=\\"tooltip\\" href=\\"/path/to\\"><img class=\\"thumb\\" src=\\"https://example.com/img.jpg\\" alt=\\"Alt\\" width=\\"100px\\" height=\\"100px\\">Label</a>"`
  );
  unmountComponentAtNode(container);
});

it("renders without crashing whithout optional props", () => {
  const container = document.createElement("div");
  act(() => {
    render(
      <StaticRouter>
        <Thumbnail
          label="Label"
          imgAlt="Alt"
          imgSrc="https://example.com/img.jpg"
          to="/path/to"
        />
      </StaticRouter>,
      container
    );
  });
  expect(container.innerHTML).toMatchInlineSnapshot(
    `"<a class=\\"link\\" href=\\"/path/to\\"><img class=\\"thumb\\" src=\\"https://example.com/img.jpg\\" alt=\\"Alt\\">Label</a>"`
  );
  unmountComponentAtNode(container);
});
