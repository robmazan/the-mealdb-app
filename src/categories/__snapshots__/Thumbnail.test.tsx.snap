// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Link
  className="link"
  title="tooltip"
  to="/path/to"
>
  <img
    alt="Alt"
    className="thumb"
    height="100px"
    src="https://example.com/img.jpg"
    width="100px"
  />
  Label
</Link>
`;

exports[`renders without crashing whithout optional props 1`] = `
<Link
  className="link"
  to="/path/to"
>
  <img
    alt="Alt"
    className="thumb"
    src="https://example.com/img.jpg"
  />
  Label
</Link>
`;
